### PR TITLE
User Force Advance + Async Sessions

### PR DESCRIPTION
**Pull request by adityapawar1**
### 🦋 Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### 🦧 Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
im just making this to check the diff

## 🐸 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## 🐙 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### 🐁 Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @eshabansiya

### CODE REVIEW

File: app/participants/scenario-overview/[sessionId]/[userId]/page.tsx [hunk 1]
```diff
@@ -155,7 +164,12 @@ export default function SessionFlowPage() {
         },
         payload => {
           const newPhaseIndex = payload.new.phase_index;
-          if (newPhaseIndex != null) setPhaseInd(newPhaseIndex);
+          if (
+            payload.new.user_id != userId ||
+            payload.new.session_id != sessionIdStr
+          )
```
[the remark this answers is not in the document]
Comment (adityapawar1) [reply app/participants/scenario-overview/[sessionId]/[userId]/page.tsx]: for some reason i was getting updates from other users, i think it prob is because i had 2 tabs open on different users

**Comment by adityapawar1** [comment]
NOTE: prompt saving is being weird cause supabase prompt responses changed, but i think dylan fixed that on his branch?
the advance one more time thing is fixed, so now it says finish session instead of force advance on the last phase, but idk if thats confusing wording cause maybe the facilitator doesn't know that they the participants are still on the last phase
im not really sure what you meant by removing the title? so i didn't touch it for now cause i didnt think it made sense but its easy to remove it if thats what u wanted

File: app/participants/scenario-overview/[sessionId]/[userId]/page.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (eshabansiya) [reply app/participants/scenario-overview/[sessionId]/[userId]/page.tsx]: ok yea, when you test, you need to have one privite for the participant cause otherwise they might overlap
